geomstats/geometry/hyperbolic_space.py

```diff
@@ -187,6 +275,170 @@ def extrinsic_to_intrinsic_coords(self, point_extrinsic):
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension] in Poincare ball
+                coordinates
+
+        Returns
+        -------
+        extrinsic : array-like, shape=[n_samples, dimension + 1] in
+                     extrinsic coordinate
+        """
+        assert(point.shape[-1] == 2)
+        x, y = point[:, 0], point[:, 1]
+        x2 = point[:, 0]**2
+        den = x2 + (1+y)**2
+        x = gs.to_ndarray(x, to_ndim=2, axis=0)
+        y = gs.to_ndarray(y, to_ndim=2, axis=0)
+        x2 = gs.to_ndarray(x2, to_ndim=2, axis=0)
+        den = gs.to_ndarray(den, to_ndim=2, axis=0)
+        ball_point = gs.hstack((2*x/den, (x2+y**2-1)/den))
+        return HyperbolicSpace._ball_to_extrinsic_coordinates(ball_point)
+
+    @staticmethod
+    def _extrinsic_to_half_plane_coordinates(point):
+        """
+        Convert the parameterization of a point on the Hyperbolic space
+        from its intrinsic coordinates, to the poincare upper half plane
+        coordinates.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension + 1] in intrinsic
+                coordinates
+
+        Returns
+        -------
+        point_half_plane : array-like, shape=[n_samples, dimension] in
+                           poincare ball coordinates
+        """
+        point_ball = \
+            HyperbolicSpace._extrinsic_to_ball_coordinates(point)
+        assert(point_ball.shape[-1] == 2)
+        x, y = point_ball[:, 0], point_ball[:, 1]
+        x2 = point_ball[:, 0]**2
+        den = x2 + (1-y)**2
+
+        x = gs.to_ndarray(x, to_ndim=2, axis=0)
+        y = gs.to_ndarray(y, to_ndim=2, axis=0)
+        x2 = gs.to_ndarray(x2, to_ndim=2, axis=0)
+        den = gs.to_ndarray(den, to_ndim=2, axis=0)
+
+        return gs.hstack(((2*x)/den, (1-x2-y**2)/den))
+
+    def to_coordinates(self, point, to_point_type='ball'):
+        """
+        Convert the parameterization of a point on the Hyperbolic space
+        from current coordinates system to the coordinates system given
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dimension] expected or
+                shape=[n_samples, dimension + 1] for extrinsic
+                coordinates only
+
+        to_point_type : coordinates type to transform the point, can be
+                        'ball', 'extrinsic', 'intrinsic', 'half_plane'
+
+        Returns
+        -------
+        point_to : array-like, shape=[n_samples, dimension + 1] or
+                   shape=[n_sample, dimension]
+        """
+        point = gs.to_ndarray(point, to_ndim=2, axis=0)
+        if self.point_type == to_point_type:
+            return point
+        else:
+            extrinsic = self.transform_to[
+                self.point_type+"-extrinsic"
+                ](point)
+            return self.transform_to[
+                "extrinsic-"+to_point_type
+                ](extrinsic)
```

Comment:
Ah, I understand the dict of methods now.

I guess you do not do:
```
def coordinates_change(point, from_type, to_type)
```
because all the combinasons of change of coords are not implemented yet?